genoml/preprocessing/munging.py

```diff
@@ -15,340 +15,369 @@
-                        subprocess.run(cmd, shell=True)
-
-                if (self.gwas_path != None) & (self.geno_path != None):
-                    print(
-                        f"A list of variants and the allele being counted in the dosages (usually the minor allele) can be found here: {self.run_prefix}.variants_and_alleles.tab")
-                    for cmd in cmds_b:
-                        subprocess.run(cmd, shell=True)
-
-        if (self.geno_path != None):
-            
-            g = read_plink1_bin('temp_genos.bed')
-            g_pruned = g.drop(['fid','father','mother','gender', 'trait', 'chrom', 'cm', 'pos','a1'])
-
-            g_pruned = g_pruned.set_index({'sample':'iid','variant':'snp'})
-            g_pruned.values = g_pruned.values.astype('int')
-
-        # swap pandas-plink genotype coding to match .raw format...more about that below:
-
-        # for example, assuming C in minor allele, alleles are coded in plink .raw labels homozygous for minor allele as 2 and homozygous for major allele as 0:
-        #A A   ->    0   
-        #A C   ->    1   
-        #C C   ->    2
-        #0 0   ->   NA
-
-        # where as, read_plink1_bin flips these, with homozygous minor allele = 0 and homozygous major allele = 2
-        #A A   ->    2   
-        #A C   ->    1   
-        #C C   ->    0
-        #0 0   ->   NA
-
-            two_idx = (g_pruned.values == 2)
-            zero_idx = (g_pruned.values == 0)
-
-            g_pruned.values[two_idx] = 0
-            g_pruned.values[zero_idx] = 2
-
-            g_pd = g_pruned.to_pandas()
-            g_pd.reset_index(inplace=True)
-            raw_df = g_pd.rename(columns={'sample': 'ID'})
-        #    del raw_df.index.name
-        #    del raw_df.columns.name
-            
-        # now, remove temp_genos
+        self.pheno_df.to_hdf(self.output_datafile, key="pheno", mode="w")
+        addit_df = None
+
+        if self.geno_path:
+            self._run_external_plink_commands()
+
+            # read_plink1_bin reads in non-reference alleles. When `ref="a1"` (default),
+            # it counts "a0" alleles. We want to read homozygous minor allele as 2 and
+            # homozygous for major allele as 0. ref="a0" ensures we are counting the
+            # homozygous minor alleles. This matches the .raw labels.
+            g = read_plink1_bin("temp_genos.bed", ref="a0")
+            g = g.drop(
+                [
+                    "fid",
+                    "father",
+                    "mother",
+                    "gender",
+                    "trait",
+                    "chrom",
+                    "cm",
+                    "pos",
+                    "a1",
+                ]
+            )
+
+            g = g.set_index({"sample": "iid", "variant": "snp"})
+
+            self.merged = g.to_pandas()
+            del g
+            self.merged.reset_index(inplace=True)
+            self.merged = self.merged.rename(columns={"sample": "ID"}, inplace=True)
```

Comment:
Do you have to still do the assignment when `inplace=True`, is that redundant?

Reply:
Also maybe just a personal style thing, but I think it might make more sense to store this as a local variable, and only set it to `self.merged` when it is finished?  

I like to keep variable local as long as possible generally, since it makes the code easier to follow (no chance of external influence on a purely local variable, while as soon as I see it assigned to `self` I have to worry where it might be going, and whether that matters yet.

Reply:
good catch, inplace does not need a return.

Reply:
Also, the `self.merged` came about because there was a lot more variables in this file previously (which I removed so they were unified).

I am going to split this into a separate function.